python/pyspark/serializers.py

```diff
@@ -296,7 +296,7 @@ def create_array(s, t):
             mask = s.isnull()
             # Ensure timestamp series are in expected form for Spark internal representation
             if t is not None and pa.types.is_timestamp(t):
-                s = _check_series_convert_timestamps_internal(s.fillna(0), self._timezone)
+                s = _check_series_convert_timestamps_internal(s, self._timezone)
                 # TODO: need cast after Arrow conversion, ns values cause error with pandas 0.19.2
```

Comment:
This looks related to pandas 0.19.2 too. Should we deal with it?

Reply:
Let's fix this one up in another PR.

Reply:
I fixed this here as well since it's pretty small. I am happy to revert and make it a seperate PR if we prefer that. Please let me know :)